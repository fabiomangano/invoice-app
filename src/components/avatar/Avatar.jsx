import React from 'react';
import PropTypes from 'prop-types';
import placeholderAvatar from '../../assets/image-avatar.jpg';
import './Avatar.style.scss';

const Avatar = ({ src, size, alt }) => (
  <img className="avatar" src={src} alt={alt} size={size} />
);

Avatar.defaultProps = {
  size: 'medium',
  src: placeholderAvatar,
};

Avatar.propTypes = {
  src: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  alt: PropTypes.string.isRequired,
};

export default Avatar;
