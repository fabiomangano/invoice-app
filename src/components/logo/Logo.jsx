import React from 'react';
import { ReactComponent as LogoIcon } from '../../assets/logo.svg';
import './Logo.style.scss';

const Logo = () => (
  <div className="logo">
    <div className="logo__top-box" />
    <div className="logo__bottom-box" />
    <LogoIcon className="logo__icon" />
  </div>
);

export default Logo;
