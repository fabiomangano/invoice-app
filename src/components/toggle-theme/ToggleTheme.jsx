import React from 'react';
import PropTypes from 'prop-types';
import { ReactComponent as LightModeIcon } from '../../assets/icon-moon.svg';
import { ReactComponent as DarkModeIcon } from '../../assets/icon-sun.svg';
import './ToggleTheme.style.scss';

const ToggleTheme = ({ theme, setTheme }) => (
  <button className="toggle-theme" type="button" onClick={setTheme}>
    {theme === 'light' ? <DarkModeIcon /> : <LightModeIcon />}
  </button>
);

ToggleTheme.propTypes = {
  theme: PropTypes.oneOf(['light', 'dark']).isRequired,
  setTheme: PropTypes.func.isRequired,
};

export default ToggleTheme;
