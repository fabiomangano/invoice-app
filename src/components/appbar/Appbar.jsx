import React from 'react';
import Avatar from '../avatar/Avatar';
import ToggleTheme from '../toggle-theme/ToggleTheme';
import Logo from '../logo/Logo';
import './Appbar.style.scss';

const Appbar = () => (
  <div className="appbar">
    <Logo />
    <div className="appbar__menu">
      <div className="appbar__option">
        <ToggleTheme theme="dark" setTheme={() => {}} />
      </div>
      <hr className="appbar__divider" />
      <div className="appbar__option">
        <Avatar alt="Profile img" />
      </div>
      <hr className="appbar__divider" />
    </div>
  </div>
);

export default Appbar;
