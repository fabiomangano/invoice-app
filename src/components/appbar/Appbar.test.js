import React from 'react';
import Renderer from 'react-test-renderer';
import Appbar from './Appbar';
import Avatar from '../avatar/Avatar';
import Logo from '../logo/Logo';
import ToggleTheme from '../toggle-theme/ToggleTheme';

jest.mock('../avatar/Avatar', () => () => <svg />);
jest.mock('../logo/Logo', () => () => <svg />);
jest.mock('../toggle-theme/ToggleTheme', () => () => (
  <button type="button">toggleTheme</button>
));

describe('Appbar', () => {
  let tree;
  let theme;
  let toggleTheme;

  beforeAll(() => {
    theme = 'light';
    toggleTheme = jest.fn();
  });

  afterEach(() => {
    tree = null;
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should render an horizontal Appbar', () => {
    Renderer.act(() => {
      tree = Renderer.create(
        <Appbar variant="horizontal">
          <Logo />
          <Appbar.Menu>
            <Appbar.Option>
              <ToggleTheme theme={theme} toggleTheme={toggleTheme} />
            </Appbar.Option>
            <Appbar.Divider />
            <Appbar.Option>
              <Avatar alt="Profile Image" />
            </Appbar.Option>
          </Appbar.Menu>
        </Appbar>,
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render a vertical Appbar', () => {
    Renderer.act(() => {
      tree = Renderer.create(
        <Appbar variant="vertical">
          <Logo />
          <Appbar.Menu>
            <Appbar.Option>
              <ToggleTheme theme={theme} toggleTheme={toggleTheme} />
            </Appbar.Option>
            <Appbar.Divider />
            <Appbar.Option>
              <Avatar alt="Profile Image" />
            </Appbar.Option>
          </Appbar.Menu>
        </Appbar>,
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
