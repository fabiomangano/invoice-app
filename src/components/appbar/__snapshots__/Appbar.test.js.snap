// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appbar should render a vertical Appbar 1`] = `
<div
  className="appbar appbar--vertical"
>
  <svg />
  <div
    className="menu menu--vertical"
  >
    <div
      className="option option--vertical"
    >
      <button
        type="button"
      >
        toggleTheme
      </button>
    </div>
    <hr
      className="divider divider--vertical"
    />
    <div
      className="option option--vertical"
    >
      <svg />
    </div>
  </div>
</div>
`;

exports[`Appbar should render an horizontal Appbar 1`] = `
<div
  className="appbar appbar--horizontal"
>
  <svg />
  <div
    className="menu menu--horizontal"
  >
    <div
      className="option option--horizontal"
    >
      <button
        type="button"
      >
        toggleTheme
      </button>
    </div>
    <hr
      className="divider divider--horizontal"
    />
    <div
      className="option option--horizontal"
    >
      <svg />
    </div>
  </div>
</div>
`;
